src/test/java/org/mars/rover/kata/KataTest.java

```diff
@@ -0,0 +1,99 @@
+package org.mars.rover.kata;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.util.Scanner;
+
+public class KataTest {
+    @Test
+    public void testRoverClassExists() {
+        String className = "org.mars.rover.kata.MarsRover";
+
+        try {
+            Class<?> myClass = Class.forName(className);
+            Assertions.assertNotNull(myClass);
+            Assertions.assertEquals(className, myClass.getName());
+            System.out.println("Class " + className + " exists.");
+        } catch (ClassNotFoundException e) {
+            Assertions.fail("Class " + className + " does not exist.");
+        }
+    }
+
+    @Test
+    public void testRoverReceivesStartingPointAndPosition() {
+        MarsRover marsRover = new MarsRover();
+
+        marsRover.setDirection(Direction.N);
+        marsRover.setCoordinate(new Coordinate(0, 1));
+
+        Assertions.assertEquals(marsRover.getDirection(), Direction.N);
+        Assertions.assertEquals(marsRover.getCoordinate().getX(), 0);
+        Assertions.assertEquals(marsRover.getCoordinate().getY(), 1);
+    }
+
+    @Test
+    public void testNavigatorClassExists() {
+        String className = "org.mars.rover.kata.MarsNavigator";
+
+        try {
+            Class<?> myClass = Class.forName(className);
+            Assertions.assertNotNull(myClass);
+            Assertions.assertEquals(className, myClass.getName());
+            System.out.println("Class " + className + " exists.");
+        } catch (ClassNotFoundException e) {
+            Assertions.fail("Class " + className + " does not exist.");
+        }
+    }
+
+    @Test
+    public void testNavigatorConstructsUpperRightCoordinates() {
+        String coordinatesString = "5 5";
+        System.setIn(new ByteArrayInputStream(coordinatesString.getBytes()));
+        MarsNavigator marsNavigator = new MarsNavigator(new Scanner(System.in));
+
+        marsNavigator.loadInput();
```

Comment:
O proszę i tutaj dochodzimy do sedna sprawy. Czytając ten kod widać, że klasa `MarsNavigator`, wbrew nazwie, zajmuje się nie tylko nawigacją marsjańską, ale też pobieraniem danych (metoda `loadInput`) czy parsowaniem komend `processInput`.

Zgodnie z [Single Responsibility Principle (SRP)](https://en.wikipedia.org/wiki/Single-responsibility_principle) najlepiej by było, aby klasa zajmowała się tylko jedną rzeczą naraz. Ma to wiele zalet, ale najłatwiejsze do zauważenia już w Twoim kodzie są:
 * klasy są mniejsze i łatwe do zrozumienia
 * dużo łatwiej jest testować klasy zgodnie z SRP, bo można im wprost podać dane wejściowe bez ustawiania całego środowiska, tak jak np. Ty tutaj wstrzykujesz komendy wejściowe do `System.in`.
 
Szczególnie porządane przez swoją wygodę jest tworzenie klas z tzw. [pure functions](https://dzone.com/articles/pure-bliss-with-pure-functions-in-java), czyli funkcjami które na zadane dane wejściowe zawsze zwracają ten sam wynik.

Myślę, że warto abyśmy zrefaktorowali ten kod razem na najbliższym spotkaniu, bo wtedy zobaczysz z pierwszej ręki jak praktycznie rozdzielić od siebie odpowiedzialności i jak w wyniku tego zmienią się klasy testów.

Reply:
Dodane.